ml_pipeline/readme.md

```diff
@@ -0,0 +1,134 @@
+# Machine Learning Pipeline for Raster Data
+
+This repository contains a machine learning pipeline for working with raster data. The pipeline includes scripts for generating training samples, training a model, making predictions, and more. Below, you'll find information on the folder structure and how to use each script.
+
+## Folder Structure
+
+Your project directory should have the following structure:
+
+```
+ml_pipeline/
+│
+├── pickles_train_latest/
+│   ├── band1.pkl
+│   ├── band2.pkl
+│   └── ...
+│
+├── pretrained_model/
+│   ├── model.pkl
+│   └── ...
+│
+├── scripts/
+│   ├── generate_training_sample.py
+│   ├── sample_raster.py
+│   ├── training.py
+│   └── prediction.py
+│
+├── config.json
+├── README.md
+└── requirements.txt
+```
+
+- **`pickles_train_latest/`**: This folder contains `.pkl` files for each band for the purpose of rescaling.
+- **`pretrained_model/`**: This folder contains a pre-trained classification model.
+- **`scripts/`**: This folder contains all the scripts needed for the pipeline.
+  - **`generate_training_sample.py`**: Generates point shapefiles for each class from the label raster for training.
+  - **`sample_raster.py`**: Uses input from `generate_training_sample.py` and rasterizes each point using the input stacked raster.
+  - **`training.py`**: Used to train a model on the data generated.
+  - **`prediction.py`**: Makes predictions using a pre-trained model and takes a config file as input.
+
+
+## Prerequisites
+
+1. Python 3.10.0
+2. Required Python packages:
+   - GDAL=3.5.0
+   - json
+   - numpy
+   - rasterio
+   - geopandas
+   - pycaret
+   - tqdm
+
+You can install the required packages using the following command:
+```bash
+pip install numpy rasterio geopandas pycaret tqdm
+```
+
+Sure, I can help you update your README file to reflect the folder structure and provide clear instructions for running your pipeline. Here's your updated README:
```

Comment:
Did chatgpt write this readme? You might want to delete this line :wink:

Reply:
Yes, 
I accidently deleted my readme :P
I have fixed it :D